mkosi/__init__.py

```diff
@@ -3303,6 +3303,12 @@ def copy_git_files(src: Path, dest: Path, *, source_file_transfer: SourceFileTra
         files |= {sm / x.decode("utf-8") for x in c.stdout.rstrip(b"\0").split(b"\0")}
         files -= submodules
 
+        # Add the .git submodule file well.
+        if source_file_transfer == SourceFileTransfer.copy_git_more:
+            fp = os.path.join(str(src / sm), ".git")
+            fr = fp[len(str(src))+1 :]
+            files.add(fr)
```

Comment:
The first line of this block would be better as
```python
fp = src / sm / ".git"
```
but I gather you just want
```suggestion
            files.add(os.path.join(sm, ".git"))
```
or
```suggestion
            files.add(f"{sm}/.git")
```
no?

Reply:
Ah, indeed. I went with the first suggestion.